packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -87,6 +87,62 @@ class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasse
 
         return super.generateClassNames(classes);
     }
+
+    /**
+     * Generates the correct attributes
+     */
+    private generateAttributes(): object {
```

Comment:
I'd like to get away from the "generateAttributes" pattern - it was mainly implemented as a way to work around inaccurate typings and it was always a bit confusing as to when an attribute would be added in the render function and when it would be added via the `generateAttributes` method. I don't think you'll really need this though, with the other changes.